OpenRA.Mods.Common/Widgets/RadarWidget.cs

```diff
@@ -372,7 +375,9 @@ public override void Tick()
 							if (!t.Actor.IsInWorld || world.FogObscures(t.Actor))
 								continue;
 
-							foreach (var cell in t.Trait.RadarSignatureCells(t.Actor))
+							cells.Clear();
+							t.Trait.RadarSignatureCells(t.Actor, cells);
+							foreach (var cell in cells)
```

Comment:
Can we pull the actual cell drawing outside the parent `foreach` and skip `cells.Clear()` altogether?
Like 
`foreach (var t in world.ActorsWithTrait<IRadarSignature>()) { /* populate all cells for all actors */ }`
and then 
`foreach (var cell in cells) { /* draw everything in one go */ }`.
Then we'd only need to clear the list at the start/end of the tick.

Reply:
As you note below, this would require the buffer to resize until it can hold the whole set of signatures. The benefit of the current loop is that the buffer will only need to be big enough to hold the largest signature.